### PR TITLE
fix channel search and channel detail

### PR DESCRIPTION
**Pull request by yustinuspr**
Fix channel search and channel detail to use helix twitch API.

This is still untested though. I unable to test it.
It might be good if you give some steps to test it so I can be more asure :)

### CODE REVIEW

**Comment by fgl27** [comment]
You didn't fix searches you also change things that may bricks the app.

If you don't know how to test why don't you ask for that first?

**AddCode_client_token** is the app security key

**AddCode_main_token** is the app generated bearer token

Just create a app on twitch website replace the client id and secret and the app will gen the main token and show content on live section, after that you can test API that work with app token.

**Comment by yustinuspr** [comment]
I've tried to test the API myself before creating this PR. And I've updated some part to follow new API

I've also done some step to test hoping that I can open it as website just like your [github.io](https://fgl27.github.io/SmartTwitchTV/release/index.html) by doing these:
1. Change `AddCode_clientId` and `AddCode_client_token` to my app client ID and token
2. Create release by running `./release/release_maker.sh`
3. Open the `release/master.js` using node `node release/master.js`
4. All I got was error `Main_Start` is not defined

**Comment by fgl27** [comment]
To test the app you open the html file from app folder, is that simple.

File: app/specific/ScreensObj.js [hunk 1]
```diff
@@ -1235,11 +1235,12 @@ function ScreensObj_InitSearchChannels() {
...
         screen: Main_SearchChannels,
         object: 'channels',
-        base_url: Main_kraken_api + 'search/channels?limit=' + Main_ItemsLimitMax + '&query=',
+        use_helix: true,
+        base_url: Main_helix_api + 'search/channels?first=' + Main_ItemsLimitMax + '&query=',
         set_url: function() {
             if (this.offset && (this.offset + Main_ItemsLimitMax) > this.MaxOffset) this.dataEnded = true;
             this.url = this.base_url + encodeURIComponent(Main_values.Search_data) +
-                '&offset=' + this.offset;
+                '&after=' + this.offset; // offset probably still broken, the behaviour changed
```
Comment (zweed4u) [inline app/specific/ScreensObj.js:1243]: This should look something like:
```
if (this.use_helix === false && this.offset && (this.offset + Main_ItemsLimitMax) > this.MaxOffset) this.dataEnded = true;
this.url = this.base_url + encodeURIComponent(Main_values.Search_data) + (this.cursor ? '&after=' + this.cursor : '');
```

File: app/specific/ScreensObj.js [hunk 2]
```diff
@@ -1235,11 +1235,12 @@ function ScreensObj_InitSearchChannels() {
         table: 'stream_table_search_channel',
         screen: Main_SearchChannels,
         object: 'channels',
-        base_url: Main_kraken_api + 'search/channels?limit=' + Main_ItemsLimitMax + '&query=',
+        use_helix: true,
+        base_url: Main_helix_api + 'search/channels?first=' + Main_ItemsLimitMax + '&query=',
```
Comment (zweed4u) [inline app/specific/ScreensObj.js:1239]:
```suggestion
        base_url: `${Main_helix_api}search/channels?first=${Main_ItemsLimitMax}&query=`,
```

Comment (yustinuspr) [reply app/specific/ScreensObj.js:1251]: This gives error `'template literal syntax' is only available in ES6 (use 'esversion: 6')`. I ignore this for the moment.

**Comment by yustinuspr** [comment]
@fgl27 I updated the pull reverting the thing you said bricked the app. 

In this patch, I able to test by myself:
 - search channel
 - Open channel detail (some detail still missing)
 - Open VODS list
 - Clicking one the vods

But I **unable** to test the player because I got CORS error for _usher_ request.

Hopefully you can help me test it

**Comment by yustinuspr** [comment]
@TheMY3 That's the goal. Although, the VODs that I fixed is the one on Channel's detail (you should search the channel first). Not the `Videos` one on the sidebar

**Comment by TheMY3** [comment]
@yustinuspr can you help with release? I install all npm dependencies, run `./smarttv-twitch/release/release_maker.sh` and go large output with errors like:
```
Error: ENOENT: no such file or directory, open './smarttv-twitch/release/temp_maker/SettingsLanguages.js'
    at Object.openSync (fs.js:498:3)
    at Object.writeFileSync (fs.js:1524:35)
    at run (/usr/local/lib/node_modules/uglify-js/bin/uglifyjs:467:12)
    at Object.<anonymous> (/usr/local/lib/node_modules/uglify-js/bin/uglifyjs:294:5)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47 {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: './smarttv-twitch/release/temp_maker/SettingsLanguages.js'
}
./smarttv-twitch/release/release_maker.sh: line 130: ./smarttv-twitch/release/master.js: No such file or directory
	Including compresed version of Sidepannel.js to master.js
internal/fs/utils.js:314
    throw err;
    ^
```

And that wa for large amount of files, but file generated, is it ok or release archive should be broken?
image

**Comment by yustinuspr** [comment]
@TheMY3 I believe you're using linux. You supposed to run it from user home. You shall use `~`

I also encounter this problem at first that made me unable to test the app

**Comment by TheMY3** [comment]
> @TheMY3 I believe you're using linux. You supposed to run it from user home. You shall use `~`
> 
> I also encounter this problem at first that made me unable to test the app

Yeah, I run it first from Arch and then from Ubuntu, ok, should try it later

**Comment by fgl27** [comment]
After the latest twitch change they made so you can't share the app keys meaning what is on the repo will not work unless you add yours own keys + some twitch public keys so the player can work (I will explain in another post how to add keys later or after a day or two, I need to be on my personal computer to get all the info).

But even if you do all that the VOD player is not updated may not play a thing or play but not load chat and any info about the VOD.

Plus pagination on VOD and Channel screen isn't fixed on this commit, if you scroll down the app may crash or freeze

When all of that is OK I will merge myself this change, is just too much and I’m busy with my personal life.

**Comment by yustinuspr** [comment]
> @fgl27: When all of that is OK I will merge myself this change, is just too much and I’m busy with my personal life.
We appreciate for all of your work in this app. As users, we want to help you by fixing the app and use it better. Hopefully this pull request won't burden you more.

> @fgl27: Plus pagination on VOD and Channel screen isn't fixed on this commit, if you scroll down the app may crash or freeze
Should be fixed in my latest push. Thank you for the input.

**Comment by fgl27** [comment]
thanks you!

**Comment by TheMY3** [comment]
@fgl27 can you release new version please?

**Comment by fgl27** [comment]
I already push this change to the app

**Comment by TheMY3** [comment]
> I already push this change to the app

Thank you, it is working now :tada:

**Comment by fgl27** [comment]
when I push a change like this, I forget to inform you need to close the app and reopen for the changes to show.

To fully close it, you can just hold the return key when the app is open and in use.

**Pull request merged**